nulib/meadow
# Update MIME lambda to grab first 1K of file from S3

# Summary 

Uses a 1K buffer (instead of a Readable Stream or Tokenizer) to determine the MIME type of the file. Since the `file-type` package uses file signatures, and the magic number is always within the first 512 bytes of the file, this seems like a reasonable approach, and also has the advantage of putting our function (as opposed to the `file-type` or `aws-sdk` code) in charge of scoping and garbage collection.

This change requires a terraform apply, which has already been done in the dev environment but not staging or prod.

# Specific Changes in this PR
- Update `lambdas/mime_type/index.js` to use a 1K buffer instead of a readable stream

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.


# Steps to Test

Either:
- Run a large (500+ file set) ingest sheet of wav, jpg, and xml files through the dev environment pipeline
- Write a script to throw 500+ wav, jpg, and xml files at the `dev-environment-mime-type` lambda in quick succession and make sure they all return sane-looking values

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
- Backwards-incompatible API changes


# Tested/Verified
